src/main/java/hudson/plugins/templateproject/ItemListenerImpl.java

```diff
@@ -47,11 +62,46 @@ public void onLoaded() {
 		}
 	}
 
+	@Override
+	public void onRenamed(Item item, String oldName, String newName) {
+		for (AbstractProject project : hudson.getAllItems(AbstractProject.class)) {
```

Comment:
I would leave it as is now and create a reference management engine later. There are project actions, which allow to persist and manage the data. This solution will be pretty complex, but useful